cms/djangoapps/contentstore/management/commands/backfill_orgs_and_org_courses.py

```diff
@@ -0,0 +1,168 @@
+        orgslug_coursekey_pairs = find_orgslug_coursekey_pairs()
+        orgslug_library_pairs = find_orgslug_library_pairs()
+        orgslugs = (
+            {orgslug for orgslug, _ in orgslug_coursekey_pairs} |
+            {orgslug for orgslug, _ in orgslug_library_pairs}
+        )
+        orgs = [
+            {"short_name": orgslug, "name": orgslug}
+            for orgslug in sorted(orgslugs)
+        ]
+        org_coursekey_pairs = [
+            ({"short_name": orgslug}, coursekey)
+            for orgslug, coursekey in sorted(orgslug_coursekey_pairs)
+        ]
+
+        # Note that edx-organizations code will handle:
+        # * Not overwriting existing organizations.
+        # * Skipping duplicates, based on the short name (case-insensiive).
+
+        # Start with a dry run.
+        # This will log to the user which orgs/org-courses will be created/reactivated.
+        organizations_api.bulk_add_organizations(
+            orgs,
+            dry_run=True,
+        )
+        organizations_api.bulk_add_organization_courses(
+            org_coursekey_pairs,
+            dry_run=True,
+        )
+        if not should_apply_changes(options):
+            print("No changes applied.")
+            return
+
+        # It's go time.
+        print("Applying changes...")
+        organizations_api.bulk_add_organizations(
+            orgs,
+            dry_run=False,
+        )
+        organizations_api.bulk_add_organization_courses(
+            org_coursekey_pairs,
+            dry_run=False,
+        )
+        print("Changes applied successfully.")
+
+
+def should_apply_changes(options):
+    """
+    Should we apply the changes to the db?
+
+    If already specified on the command line, use that value.
+
+    Otherwise, prompt.
+    """
+    if options.get('apply') and options.get('dry'):
+        raise CommandError("Only one of 'apply' and 'dry' may be specified")
+    if options.get('apply'):
+        return True
+    if options.get('dry'):
+        return False
+    answer = ""
+    while answer.lower() not in {'y', 'yes', 'n', 'no'}:
+        answer = input('Commit changes shown above to the database [y/n]? ')
+    return answer.lower().startswith('y')
+
+
+def find_orgslug_coursekey_pairs():
+    """
+    Returns the unique pairs of (organization short name, course run key)
+    from the CourseOverviews table, which should contain all course runs in the
+    system.
+
+    Returns: set[tuple[str, CourseKey]]
+    """
+    # Using a set comprehension removes any duplicate (org, course) pairs.
+    return {
+        (course_key.org, course_key)
+        for course_key
+        # Worth noting: This will load all CourseOverviews, no matter their VERSION.
+        # This is intentional: there may be course runs that haven't updated
+        # their CourseOverviews entry since the last schema change; we still want
+        # capture those course runs.
+        in CourseOverview.objects.all().values_list("id", flat=True)
```

Comment:
In large instances, is calling `objects.all()` a little dangerous?  My django's slightly rusty - because we're doing the `values_list`, is the memory footprint of this query significantly smaller than if we were instantiating models for all those CourseOverviews? Is there a version of this that can batch query to avoid yanking _all_ of them out of the DB simultaneously?

Reply:
For the scale of courses we have (~15K), doing it in one swoop like this with `values_list` on the `id` is totally fine. It's only going to do a scan of the index, off a table that's extremely frequently used and cached.

```
mysql> select id from course_overviews_courseoverview;
# -- lots of results here --
15499 rows in set (0.01 sec)
```